kubetest2/kubetest2-gce/deployer/common.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deployer
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"k8s.io/klog"
+	"sigs.k8s.io/boskos/client"
+	boskosCommon "sigs.k8s.io/boskos/common"
+)
+
+func (d *deployer) init() error {
+	if d.commonOptions.ShouldBuild() {
+		if err := d.verifyBuildFlags(); err != nil {
+			return fmt.Errorf("init failed to check build flags: %s", err)
+		}
+	}
+
+	if d.commonOptions.ShouldUp() {
+		if err := d.verifyUpFlags(); err != nil {
+			return fmt.Errorf("init failed to verify flags for up: %s", err)
+		}
+
+		if d.GCPProject == "" {
+			klog.Info("No GCP project provided, acquiring from Boskos")
+
+			if err := d.getProjectFromBoskos(); err != nil {
```

Comment:
I think the defaulting will need some more thought here.
for example for local testing I would think that the already set gcloud default project would be the default.

Maybe it's worth differentiating the logic for local vs CI?

Reply:
I'm not sure how I feel about defaulting to the gcloud default, because its not clear behavior to the user if they don't inspect the flags. Because kubetest2 isn't actually an extension of gcloud in any way (it just happens to use it) I think its an unclear default. Happy to discuss further, we could also consider a `--local` flag but I don't like the idea of that because it could lead to some messy divergences of behavior in the future in the name of slightly better UX.

Reply:
the gcloud default is honored by kube-up so was mainly coming from extending that. But it's also fine to have our own prescriptive required flag.

For CI vs local, it should motly be auto-detect based on what envs prow sets. Mainly PROW_JOB_ID.

Reply:
I'm favoring a prescriptive flag like we are planning for all configuration of the deployer. It may be that the deployer will have to eventually detect or be told that is or is not running in Prow, but I don't think its necessary right now.